### PR TITLE
Add XHR response decoding test cases

### PR DESCRIPTION
**Pull request by yutakahirano**
- Add test cases for 'text' response type
 - Add test cases for 'document' response type

### CODE REVIEW

**Comment by yutakahirano** [comment]
@annevk @tyoshino can you take a look?

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/19182)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/283582660)

**Review by annevk — COMMENTED** [review]
Looks good apart from the addition of a.txt and b.txt. I'd like @wisniewskit to take a look as well as he's been fixing Firefox in this area.

**Comment by yutakahirano** [comment]
Oops, sorry for adding the text files. I've just removed them.

File: XMLHttpRequest/responsetext-decoding.htm [hunk 1]
```diff
@@ -1,47 +1,89 @@
...
+      }
+      function xml(content) {
+        return "<?xml version='1.0' encoding='windows-1252'?><x>" + content + "</x>";
+      }
+      function encoded_xml(encoded_content) {
+        return encodeURIComponent("<?xml version='1.0' encoding='windows-1252'?><x>") + encoded_content + encodeURIComponent("<\/x>");
+      }
       function request(type, input, output, responseType) {
-        var test = async_test(document.title + " (" + type + " " + input + ")");
+        var test = async_test(document.title + " (" + type + " " + input + " " + (responseType ? " " + responseType : "empty") + ")");
```
Comment (tyoshino) [inline XMLHttpRequest/responsetext-decoding.htm]: var -> const

File: XMLHttpRequest/responsetext-decoding.htm [hunk 2]
```diff
@@ -1,47 +1,89 @@
...
         test.step(function() {
-          var client = new XMLHttpRequest()
+          var client = new XMLHttpRequest();
```
Comment (tyoshino) [inline XMLHttpRequest/responsetext-decoding.htm]: var -> const

File: XMLHttpRequest/responsetext-decoding.htm [hunk 3]
```diff
@@ -1,47 +1,89 @@
...
     <link rel="help" href="https://xhr.spec.whatwg.org/#the-responsetext-attribute" data-tested-assertations="following::ol[1]/li[4]" />
     <link rel="help" href="https://xhr.spec.whatwg.org/#text-response-entity-body" data-tested-assertations="following::ol[1]/li[2] following::ol[1]/li[3] following::ol[1]/li[4] following::ol[1]/li[5]" />
   </head>
   <body>
-    <div id="log"></div>
     <script>
+      function html(content) {
+        return "<!doctype html><meta charset=windows-1252><x>" + content + "</x>";
+      }
+      function encoded_html(encoded_content) {
```
Comment (tyoshino) [inline XMLHttpRequest/responsetext-decoding.htm]: ditto

File: XMLHttpRequest/responsetext-decoding.htm [hunk 4]
```diff
@@ -1,47 +1,89 @@
...
     <title>XMLHttpRequest: responseText decoding</title>
     <script src="/resources/testharness.js"></script>
     <script src="/resources/testharnessreport.js"></script>
     <link rel="help" href="https://xhr.spec.whatwg.org/#the-responsetext-attribute" data-tested-assertations="following::ol[1]/li[4]" />
     <link rel="help" href="https://xhr.spec.whatwg.org/#text-response-entity-body" data-tested-assertations="following::ol[1]/li[2] following::ol[1]/li[3] following::ol[1]/li[4] following::ol[1]/li[5]" />
   </head>
   <body>
-    <div id="log"></div>
     <script>
+      function html(content) {
```
Comment (tyoshino) [inline XMLHttpRequest/responsetext-decoding.htm]: how about adding create_ or build_ to the function name?

File: XMLHttpRequest/responsetext-decoding.htm [hunk 5]
```diff
@@ -1,47 +1,89 @@
...
+        return "<!doctype html><meta charset=windows-1252><x>" + content + "</x>";
+      }
+      function encoded_html(encoded_content) {
+        return encodeURIComponent("<!doctype html><meta charset=windows-1252><x>") + encoded_content + encodeURIComponent("<\/x>");
+      }
+      function xml(content) {
```
Comment (tyoshino) [inline XMLHttpRequest/responsetext-decoding.htm]: ditto

File: XMLHttpRequest/responsetext-decoding.htm [hunk 6]
```diff
@@ -1,47 +1,89 @@
...
+        return "<?xml version='1.0' encoding='windows-1252'?><x>" + content + "</x>";
+      }
+      function encoded_xml(encoded_content) {
```
Comment (tyoshino) [inline XMLHttpRequest/responsetext-decoding.htm]: ditto

File: XMLHttpRequest/responsedocument-decoding.htm [hunk 7]
```diff
@@ -0,0 +1,41 @@
+<!doctype html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>XMLHttpRequest: response document decoding</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <link rel="help" href="https://xhr.spec.whatwg.org/#the-responsetext-attribute" data-tested-assertations="following::ol[1]/li[4]" />
```
Comment (tyoshino) [inline XMLHttpRequest/responsedocument-decoding.htm]: Should this be changed to https://xhr.spec.whatwg.org/#the-response-attribute ?

File: XMLHttpRequest/responsedocument-decoding.htm [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    <link rel="help" href="https://xhr.spec.whatwg.org/#text-response-entity-body" data-tested-assertations="following::ol[1]/li[2] following::ol[1]/li[3] following::ol[1]/li[4] following::ol[1]/li[5]" />
```
Comment (tyoshino) [inline XMLHttpRequest/responsedocument-decoding.htm]: This anchor is gone.

**Review by tyoshino — APPROVED** [review]

File: XMLHttpRequest/responsedocument-decoding.htm (hunk 7 above)
Comment (annevk) [reply XMLHttpRequest/responsedocument-decoding.htm]: We should probably just drop these or link without anchor.

File: XMLHttpRequest/responsetext-decoding.htm (hunk 4 above)
Comment (yutakahirano) [reply XMLHttpRequest/responsetext-decoding.htm]: Done.

File: XMLHttpRequest/responsetext-decoding.htm (hunk 3 above)
Comment (yutakahirano) [reply XMLHttpRequest/responsetext-decoding.htm]: Done.

File: XMLHttpRequest/responsetext-decoding.htm (hunk 5 above)
Comment (yutakahirano) [reply XMLHttpRequest/responsetext-decoding.htm]: Done.

File: XMLHttpRequest/responsetext-decoding.htm (hunk 6 above)
Comment (yutakahirano) [reply XMLHttpRequest/responsetext-decoding.htm]: Done.

File: XMLHttpRequest/responsetext-decoding.htm (hunk 1 above)
Comment (yutakahirano) [reply XMLHttpRequest/responsetext-decoding.htm]: Removed this declaration.

File: XMLHttpRequest/responsetext-decoding.htm (hunk 2 above)
Comment (yutakahirano) [reply XMLHttpRequest/responsetext-decoding.htm]: Done.

File: XMLHttpRequest/responsedocument-decoding.htm (hunk 7 above)
Comment (yutakahirano) [reply XMLHttpRequest/responsedocument-decoding.htm]: Removed link elements.

File: XMLHttpRequest/responsedocument-decoding.htm (hunk 8 above)
Comment (yutakahirano) [reply XMLHttpRequest/responsedocument-decoding.htm]: Removed the link elements.

**Comment by wisniewskit** [comment]
I don't see any obvious problem with the tests themselves (but then I'm really not an expert on the decoder aspects of the spec).

The one issue I do see is that the manifest isn't being updated to add the new responsedocument-decoding.htm test-file. I'm not sure if that will be automatically done as the patch is merged, so I thought I'd mention it.

**Comment by annevk** [comment]
Thanks, manifests are a Mozilla-specific affair and are updated automatically I believe.

**Pull request merged**